common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1121,10 +1121,12 @@ public boolean matches(String expr) {
      */
     @Override
     public int hashCode() {
-        if (hash == 0) {
-            hash = PlatformDependent.hashCodeAscii(value, offset, length);
+        int h = hash;
```

Comment:
> it would only be problematic if we would have a constructor that directly set the hash

No. This is about a situation where two threads at the same time are call the `hashCode()` (or `toString()`) method. Theoretically, one of them can get wrong result `0` (or `null`) due to double reading from non-volatile/non-final field, because the compiler can reorder the field reads.